internal/services/cognitive/cognitive_account_connection_resource.go

```diff
@@ -0,0 +1,512 @@
+				props := model.Properties
+
+				base := props.ConnectionPropertiesV2()
+				state.AuthType = string(base.AuthType)
+				state.Category = pointer.FromEnum(base.Category)
+				state.Target = pointer.From(base.Target)
+				state.Metadata = map[string]string{}
+
+				// Only include metadata fields that were in the original config
+				if len(currentState.Metadata) > 0 {
+					apiMetadata := pointer.From(base.Metadata)
+
+					for configKey := range currentState.Metadata {
+						for apiKey, apiValue := range apiMetadata {
+							if strings.EqualFold(configKey, apiKey) {
+								state.Metadata[configKey] = apiValue
+								break
+							}
+						}
+					}
+				}
+
+				// Handle auth-specific properties
+				// Note: Sensitive credentials are not returned by API, so preserve from config
+				switch props.(type) {
+				case accountconnectionresource.ApiKeyAuthConnectionProperties:
+					state.ApiKey = currentState.ApiKey
+
+				case accountconnectionresource.OAuth2AuthTypeConnectionProperties:
+					if len(currentState.OAuth2) > 0 {
+						state.OAuth2 = []OAuth2AuthModel{{
+							ClientId:       currentState.OAuth2[0].ClientId,
+							ClientSecret:   currentState.OAuth2[0].ClientSecret,
+							AuthURL:        currentState.OAuth2[0].AuthURL,
+							DeveloperToken: currentState.OAuth2[0].DeveloperToken,
+							Password:       currentState.OAuth2[0].Password,
+							RefreshToken:   currentState.OAuth2[0].RefreshToken,
+							TenantId:       currentState.OAuth2[0].TenantId,
+							Username:       currentState.OAuth2[0].Username,
+						}}
+					}
+
+				case accountconnectionresource.CustomKeysConnectionProperties:
+					state.CustomKeys = currentState.CustomKeys
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CognitiveAccountConnectionResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.AccountConnectionResourceClient
+
+			id, err := accountconnectionresource.ParseConnectionID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model CognitiveAccountConnectionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			// since we update with PATCH method, and multiple models used for API payload, ignore this fields when no change detected.
+			if !metadata.ResourceData.HasChange("metadata") {
+				model.Metadata = map[string]string{}
+			}
+
+			if !metadata.ResourceData.HasChange("target") {
+				model.Target = ""
+			}
+
+			properties, err := expandConnectionProperties(model)
+			if err != nil {
+				return fmt.Errorf("expanding `properties`: %+v", err)
+			}
+
+			switch props := properties.(type) {
+			case accountconnectionresource.ApiKeyAuthConnectionProperties:
+				if !metadata.ResourceData.HasChange("api_key") {
+					props.Credentials = nil
```

Comment:
This assignment doesn't look correct. "props" is a non-pointer type local variable and changing it won't update the properties.